propose-freeform/propose-vote-enact-freeform.py

```diff
@@ -0,0 +1,279 @@
+    response = requests.get(url)
+    helpers.check_response(response)
+    # :get_assets__
+
+    # Debugging
+    # print("Assets:\n{}".format(
+    #    json.dumps(response.json(), indent=2, sort_keys=True)))
+
+    #####################################################################################
+    #                   G O V E R N A N C E   T O K E N   C H E C K                     #
+    #####################################################################################
+
+    # Get the identifier of the governance asset on the Vega network
+    assets = response.json()["assets"]
+    vote_asset_id = next((x["id"] for x in assets if x["details"]["symbol"] == "tVOTE"), None)
+    if vote_asset_id is None:
+        print("tVOTE asset not found on specified Vega network, please symbol name check and try again")
+        sys.exit(1)
+
+    # Request accounts for party and check governance asset balance
+    url = f"{node_url_rest}/parties/{pubkey}/accounts"
+    response = requests.get(url)
+    helpers.check_response(response)
+
+    # Debugging
+    # print("Accounts:\n{}".format(
+    #    json.dumps(response.json(), indent=2, sort_keys=True)))
+
+    voting_balance = 0
+    accounts = response.json()["accounts"]
+    for account in accounts:
+        if account["asset"] == vote_asset_id:
+            print("Found governance asset account")
+            print(account)
+            voting_balance = account["balance"]
+            break
+
+    if voting_balance == 0:
+        print(f"Please deposit tVOTE asset to public key {pubkey} and try again")
+        sys.exit(1)
+
+    #####################################################################################
+    #                          B L O C K C H A I N   T I M E                            #
+    #####################################################################################
+
+    # __get_time:
+    # Request the current blockchain time, and convert to time in seconds
+    response = requests.get(f"{node_url_rest}/time")
+    helpers.check_response(response)
+    blockchain_time = int(response.json()["timestamp"])
+    blockchain_time_seconds = int(blockchain_time / 1e9)  # Seconds precision
+    # :get_time__
+
+    assert blockchain_time > 0
+    assert blockchain_time_seconds > 0
+    print(f"Blockchain time: {blockchain_time} ({blockchain_time_seconds} seconds past epoch)")
+
+    #####################################################################################
+    #                               F R E E F O R M                                     #
+    #####################################################################################
+
+    # Step 1 propose a freeform proposal
+
+    # __prepare_propose_updateNetworkParameter:
+    # Compose a governance proposal for updating a network parameter
+    proposal_ref = f"{pubkey}-{helpers.generate_id(30)}"
+
+    # Set closing/enactment and validation timestamps to valid time offsets
```

Comment:
Is enactment time right for freeform? (Sorry - can't remember which of the times specifically aren't relevant for this type)